system/default_ad_api/src/motion.cpp

```diff
@@ -0,0 +1,151 @@
+// limitations under the License.
+
+#include "motion.hpp"
+
+#include <memory>
+#include <unordered_map>
+
+namespace default_ad_api
+{
+
+MotionNode::MotionNode(const rclcpp::NodeOptions & options)
+: Node("motion", options), vehicle_stop_checker_(this)
+{
+  stop_check_duration_ = declare_parameter("stop_check_duration", 1.0);
+  require_accept_start_ = declare_parameter("require_accept_start", false);
+  waiting_for_set_pause_ = false;
+
+  const auto adaptor = component_interface_utils::NodeAdaptor(this);
+  group_cli_ = create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
+  adaptor.init_srv(srv_accept_, this, &MotionNode::on_accept);
+  adaptor.init_pub(pub_state_);
+  adaptor.init_cli(cli_set_pause_, group_cli_);
+  adaptor.init_sub(sub_is_paused_, this, &MotionNode::on_is_paused);
+  adaptor.init_sub(sub_is_start_requested_, this, &MotionNode::on_is_start_requested);
+
+  rclcpp::Rate rate(10);
+  timer_ = rclcpp::create_timer(this, get_clock(), rate.period(), [this]() { on_timer(); });
+  change_state(State::kMoving, true);
+}
+
+void MotionNode::change_state(const State state, const bool init)
+{
+  using MotionState = autoware_ad_api::motion::State::Message;
+  static const auto mapping = std::unordered_map<State, MotionState::_state_type>(
+    {{State::kMoving, MotionState::MOVING},
+     {State::kStopping, MotionState::STOPPED},
+     {State::kStopped, MotionState::STOPPED},
+     {State::kStarting, MotionState::STARTING},
+     {State::kStarted, MotionState::STARTING}});
+
+  if (init || mapping.at(state_) != mapping.at(state)) {
+    MotionState msg;
+    msg.stamp = now();
+    msg.state = mapping.at(state);
+    pub_state_->publish(msg);
+  }
+  state_ = state;
+}
+
+void MotionNode::on_timer()
+{
+  if (state_ == State::kMoving) {
+    if (vehicle_stop_checker_.isVehicleStopped(stop_check_duration_)) {
+      change_state(State::kStopping);
+    }
+  }
+
+  if (state_ == State::kStarted) {
+    if (!vehicle_stop_checker_.isVehicleStopped(stop_check_duration_)) {
+      change_state(State::kMoving);
+    }
+  }
+
+  if (state_ == State::kStopping) {
+    if (!waiting_for_set_pause_ && cli_set_pause_->service_is_ready()) {
+      const auto req = std::make_shared<control_interface::SetPause::Service::Request>();
+      req->pause = true;
+      waiting_for_set_pause_ = true;
+      cli_set_pause_->async_send_request(req, [this](auto) { waiting_for_set_pause_ = false; });
+    }
+  }
+
+  if (state_ == State::kStarting) {
+    if (!waiting_for_set_pause_ && !require_accept_start_) {
+      const auto req = std::make_shared<control_interface::SetPause::Service::Request>();
+      req->pause = false;
+      waiting_for_set_pause_ = true;
+      cli_set_pause_->async_send_request(req, [this](auto) { waiting_for_set_pause_ = false; });
+    }
+  }
+}
+
+void MotionNode::on_is_paused(const control_interface::IsPaused::Message::ConstSharedPtr msg)
+{
+  switch (state_) {
```

Comment:
What is the reason for using `switch` instead of `if` only here?

Reply:
Other places only target some states, but here we use all states.

Reply:
Okay, thank you.